src/js/player.js

```diff
@@ -222,10 +222,17 @@ class Player extends Component {
     // Default state of video is paused
     this.addClass('vjs-paused');
 
-    // Make box use width/height of tag, or rely on default implementation
-    // Enforce with CSS since width/height attrs don't work on divs
-    this.width(this.options_['width'], true); // (true) Skip resize listener on load
-    this.height(this.options_['height'], true);
+    // Add a style element in the player that we'll use to set the width/height
+    // of the player in a way that's still overrideable by CSS, just like the
+    // video element
+    this.styleEl_ = document.createElement('style');
+    el.appendChild(this.styleEl_);
```

Comment:
> I could be wrong but I think rules like the example above aren't uncommon.

I think you're right, but in 4.x we set the width/height with inline styles, so the example definitely wouldn't work with video.js today.

To ensure the example would work we'd have to inject the style el at the top of the head, and remove the default styles from the main stylesheet. Removing the default styles means the embed code wouldn't have any dimensions (at least in IE8) until the player loads. Maybe that's not a real issue?